python/ray/serve/tests/test_api.py

```diff
@@ -364,7 +364,16 @@ def function2(_):
     client.create_backend("delete:v1", function2)
     client.set_traffic("delete_backend", {"delete:v1": 1.0})
 
-    assert requests.get("http://127.0.0.1:8000/delete-backend").text == "olleh"
+    for _ in range(10):
+        try:
+            assert requests.get(
+                "http://127.0.0.1:8000/delete-backend").text == "olleh"
+            break
+        except AssertionError:
+            time.sleep(0.5)  # wait for the traffic policy to propogate
```

Comment:
oh interesting -- this is effectively an API change, right? because this used to be sync. the change seems fine for now